server/controllers/dog.js

```diff
@@ -0,0 +1,128 @@
+const User = require("../models/User");
+const Dog = require("../models/Dog");
+const asyncHandler = require("express-async-handler");
+const ObjectId = require("mongoose").Types.ObjectId;
+
+// @route GET /dogs/:id:
+// @desc Get dog for logged in user by :id
+// @access Private
+exports.loadDog = asyncHandler(async (req, res, next) => {
+  const id = req.params.id;
```

Comment:
Optional: It is a good practice to validate the `req.params.id`

```
 if (!mongoose.Types.ObjectId.isValid(req.params.id)) {
    return res.status(400).send("Bad Request");
  }
```

Reply:
done! router.route("/:id").get(`validateId`, protect, loadDog);

Reply:
done ✔️